lib/Runtime/PlatformAgnostic/Platform/Linux/NumbersUtility.cpp

```diff
@@ -0,0 +1,267 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "Common.h"
+#include "ChakraPlatform.h"
+#include <locale.h>
+
+namespace PlatformAgnostic
+{
+namespace Numbers
+{
+    #define BufSize 256
+
+    template<class T, class N=T>
+    void ByteCopy(const T *from, N *to, const size_t max_length)
+    {
+        size_t length = 0;
+        for(;length < max_length && (to[length] = static_cast<N>(from[length])) != 0;
```

Comment:
This is a little hard to read- can you rewrite this to be a normal loop (with the copy done in the body)?